### ISSUE TITLE
[telegraf-ds] net and disk not getting info from host

### ISSUE THREAD

**Issue by duckman** [influxdata/helm-charts#162]
My net and disk input pluggins aren't getting the right data. If I cat /rootfs/proc/net/dev in the pod, I don't get the same as if I do cat /proc/net/dev from the host. Same with the disk related proc, but the point is, the mounting the chart is doing isn't working. I've read around trying to figure out whats going on. I'm guessing docker is doing something odd? Hoping someone knows more about this stuff than me.
Kubernetes version: v1.17.0
Docker version: 19.03.12
I did change the image being used to use the tag 1.15.2 because I have some arm hosts that the alpine images can't run on.

**Comment by rawkode** [comment]
@duckman Can you share the output of both `cats` ?

**Comment by duckman** [comment]
```
# cat /rootfs/proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
    lo:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
  eth0: 20343170227 33425266    0    0    0     0          0         0 94946501701 48602427    0    0    0     0       0          0
```
```
$ cat /proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
vetha0e90649: 190309098 1943080    0    0    0     0          0         0 183667631 1894939    0    0    0     0       0          0
veth81fd53d5: 2245576464 1834413    0    0    0     0          0         0 220264647 2112573    0    0    0     0       0          0
veth30b80b6d: 2147846643 3416903    0    0    0     0          0         0 1294152577 4052303    0    0    0     0       0          0
veth775495b0: 6716501863 41774331    0    0    0     0          0         0 10952300486 41945039    0    0    0     0       0          0
veth5838496: 16219493582 5043444    0    0    0     0          0         0 575953028 3541963    0    0    0     0       0          0
veth30024d3a: 97565915 1034351    0    0    0     0          0         0 92725446 1041423    0    0    0     0       0          0
veth3997e0f9: 6148423   98760    0    0    0     0          0         0  5665328   95239    0    0    0     0       0          0
veth7e1414e7: 234597890 2288030    0    0    0     0          0         0 261538952 2298511    0    0    0     0       0          0
flannel.1: 126583554476 118088335    0    0    0     0          0         0 19702940330 72572498    0  422    0     0       0          0
  eno1: 294362905256 344523648  149    1 1050    76          0   3490283 38755176212 63639972    0    0    0     0       0          0
veth115a115b: 6323646063 87141689    0    0    0     0          0         0 228456911510 159363840    0    0    0     0       0          0
vethf0b35707: 57464930  197913    0    0    0     0          0         0 30938061  343878    0    0    0     0       0          0
docker0: 16148885366 5043444    0    0    0     0          0         0 575925492 3541571    0    0    0     0       0          0
veth9c48fc9a: 17508703435 5089369    0    0    0     0          0         0 11801969181 8843138    0    0    0     0       0          0
vethaf8e3843: 1053613128 11568739    0    0    0     0          0         0 13357604574 16282517    0    0    0     0       0          0
vethedbcbd37: 3826561166 11032267    0    0    0     0          0         0 3919482610 8391412    0    0    0     0       0          0
veth4e0b2401: 9892163801 41241579    0    0    0     0          0         0 6813557756 42056577    0    0    0     0       0          0
 bond0: 361638675302 430579367  457 92540 1050   236          0   7077767 513296603762 490398798    0   11    0     0       0          0
veth634a406c: 12853518143 14640084    0    0    0     0          0         0 1103149301 11782913    0    0    0     0       0          0
vethb3dc5379: 982693805 3187902    0    0    0     0          0         0 326294693 3460357    0    0    0     0       0          0
  eno2: 67275770046 86055719  308    1    0   160          0   3587484 474541427550 426758826    0    0    0     0       0          0
veth7b48fbf3: 110398689  838070    0    0    0     0          0         0 620172967  948937    0    0    0     0       0          0
   br0: 346051004828 281576820    0 138809    0     0          0         0 494542194367 204079231    0    0    0     0       0          0
veth21cef983: 2392819771 2920092    0    0    0     0          0         0 2362132845 3011009    0    0    0     0       0          0
    lo: 623050882 4162192    0    0    0     0          0         0 623050882 4162192    0    0    0     0       0          0
veth13a77f46: 206976217 1857834    0    0    0     0          0         0 190684453 2135810    0    0    0     0       0          0
veth268ef603: 94952789298 48606485    0    0    0     0          0         0 20344533203 33427854    0    0    0     0       0          0
br-8a6695494b61:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
veth5a2d2d8c: 1235951016 15486164    0    0    0     0          0         0 2194776180 9548952    0    0    0     0       0          0
  cni0: 30615409358 100670924    0    0    0     0          0         0 167067718232 156521941    0    0    0     0       0          0
vethd52be97b: 14776956   49417    0    0    0     0          0         0  8668935   96440    0    0    0     0       0          0
```
First one is the pod of course and you can see its looks like it's it's own device and not the countless devises on the host.

**Comment by tribly** [comment]
I have the same problem

```
$ cat /rootfs/proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed                                                                 
  eth0: 1436804305 2219522    0    0    0     0          0         0 850634103 3022682    0    0    0     0       0          0                                                             
    lo:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                                 
 tunl0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
```
```
$ cat /proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed                                                                 
  eno2:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                                 
cali83dca353ff3: 3233262730 14293571    0    0    0     0          0         0 2793483728 21120896    0    0    0     0       0          0                                                 
kube-ipvs0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                             
flannel.1:  238732    3768    0    0    0     0          0         0   574191    8702    0 1921    0     0       0          0                                                              
virbr0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                                 
  cni0: 22950273  181345    0    0    0     0          0         0 55773745  195068    0    0    0     0       0          0                                                                
cali9de1314621b: 187714637390 67584045    0    0    0     0          0         0 38424629659 99585064    0    0    0     0       0          0                                              
    lo: 476478172396 1295046685    0    0    0     0          0         0 476478172396 1295046685    0    0    0     0       0          0                                                  
calic261ca34cb2: 1948341852 5415718    0    0    0     0          0         0 2017274240 5823477    0    0    0     0       0          0                                                   
calic77c8008033:       0       0    0    0    0     0          0         0    36928     597    0    0    0     0       0          0                                                        
cali3747adc9a10: 850837980 3023410    0    0    0     0          0         0 1437145936 2220077    0    0    0     0       0          0                                                    
 tunl0: 93466441799 76638990    0    0    0     0          0         0 418038725273 50630740    0    5    0     0       0          0                                                       
cali0838a2150f6: 104738575 1220495    0    0    0     0          0         0 871637475 1238661    0    0    0     0       0          0                                                     
  eno3:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                                 
cali4fe13136197: 1933708957 15271621    0    0    0     0          0         0 3988377427 15453877    0    0    0     0       0          0                                                 
virbr0-nic:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                             
cali407d8e12eb1: 3812518619 3517530    0    0    0     0          0         0 4789469225 3238450    0    0    0     0       0          0                                                   
docker0:  308687    5653    0    0    0     0          0         0 27669068   10046    0    0    0     0       0          0                                                                
  eno1: 244058316520 463725393    0 37650    0     0          0   1063765 1500836719873 1139776164    0    0    0     0       0          0                                                 
  eno4:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0                                                                 
cali816324857c1:       0       0    0    0    0     0          0         0     9208     135    0    0    0     0       0          0                                                        
cali2e26dfe02d0: 1933546181 15263620    0    0    0     0          0         0 3991924500 15499338    0    0    0     0       0          0
```

**Comment by philomory** [comment]
I'm seeing similar problems, although in my case, the way I noticed the issue is because the `disk` input is setting the `path` tag to values like `/etc/resolv.conf`, `/etc/hostname`, etc, rather than anything sane. I don't have easy SSH access to the nodes at the moment so I can't compare the output I'd get running it directly on the node, but those paths were a red flag.

**Comment by duckman** [comment]
I'm still struggling with this. In my reading I have come across security policies and by default containers aren't allowed privileged mode. https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privileged Could this be causing the issue? I have no idea how to change it to try it out.

**Comment by simon-weber** [comment]
This seems to happen because `/proc/net` and `/proc/mounts` are symlinks through the current pid:
```
$ ls -al /proc/net /proc/mounts /proc/self
lrwxrwxrwx 1 root root 11 Mar  6 23:05 /proc/mounts -> self/mounts
lrwxrwxrwx 1 root root  8 Mar  6 23:05 /proc/net -> self/net
lrwxrwxrwx 1 root root  0 Mar  3 01:52 /proc/self -> 12829
```

So, `/proc/self` uses the container's pid and telegraf gets the container's resources instead of the host's, despite the mount. The host's data is still accessible, (eg `cat /rootfs/proc/1/net/dev`), but I don't see an easy way to point telegraf at it.

The best I can come up with is using an overlay mount to modify the links inside the container, but [setting that up on k8s doesn't look straightforward](https://itnext.io/using-overlay-mounts-with-kubernetes-960375c05959).

**Comment by afausti** [comment]
I was struggling with that too. Unfortunately, the way the `telegraf-ds` Helm Chart is mounting the host filesystem doesn't seem to work.

If we mount the host filesystem following [this FAQ](https://github.com/influxdata/telegraf/blob/master/docs/FAQ.md#q-how-can-i-monitor-the-docker-engine-host-from-within-a-container) it works fine,  a cat on ` /proc/self/mounts` of a `df` inside the container shows the host information.

I have PR #348 with a proposed fixed.

**Comment by alespour** [comment]
Closed via #348
